zookeeper/core/component.py

```diff
@@ -289,14 +289,16 @@ def _list_field_strings(instance, color: bool, single_line: bool) -> Iterator[st
         except AttributeError as e:
             if field.allow_missing:
                 value = utils.missing
+                yield f"{field_name}={value}" if color else f"{field_name}={value}"
+                continue
             else:
                 raise e from None
 
         parent_instance = next(
             utils.generate_component_ancestors_with_field(instance, field_name), None
         )
         if parent_instance is not None:
-            is_inherited = base_getattr(parent_instance, field_name) == value  # type: ignore
+            is_inherited = base_getattr(parent_instance, field_name) is value  # type: ignore
```

Comment:
Could this be problematic if value is a literal value like an integer?
E.g. `0 is 1` would raise a warning: `SyntaxWarning: "is" with a literal. Did you mean "=="?`

Reply:
Fixed in 7460528

Reply:
This is actually not necessary, the warning is printed only if one of the elements is a literal, not a variable. `a is b` is fine, even if `a` and `b` are both ints/floats/whatever.

<img width="630" alt="image" src="https://user-images.githubusercontent.com/7688302/101024156-9e631b00-356b-11eb-9710-4828cff5c233.png">

So can we avoid adding the warning suppression?